google/cloud/storage/lifecycle_rule.cc

```diff
@@ -148,6 +166,38 @@ void LifecycleRule::MergeConditions(LifecycleRuleCondition& result,
       result.num_newer_versions.emplace(std::forward<std::int32_t>(tmp));
     }
   }
+  if (rhs.days_since_noncurrent_time.has_value()) {
+    if (result.days_since_noncurrent_time.has_value()) {
+      *result.days_since_noncurrent_time = (std::max)(
```

Comment:
Q: Do we need an indirection here when we don't 3 lines down?

Reply:
No, fixed here and everywhere else in this function.